src/EMNIST/EMNIST.jl

```diff
@@ -0,0 +1,276 @@
+export EMNIST
+module EMNIST
+    using DataDeps
+    using BinDeps
+    using FixedPointNumbers
+    using MAT: matopen, matread
+    using ..MLDatasets: bytes_to_type, datafile, download_dep, download_docstring
+    using ..MNIST: convert2image
+
+        const DEPNAME = "EMNIST"
+
+        """
+            download([dir]; [i_accept_the_terms_of_use])
+
+        Trigger the (interactive) download of the full dataset into
+        "`dir`". If no `dir` is provided the dataset will be
+        downloaded into "~/.julia/datadeps/$DEPNAME".
+
+        This function will display an interactive dialog unless
+        either the keyword parameter `i_accept_the_terms_of_use` or
+        the environment variable `DATADEPS_ALWAY_ACCEPT` is set to
+        `true`. Note that using the data responsibly and respecting
+        copyright/terms-of-use remains your responsibility.
+        """
+        download(args...; kw...) = download_dep(DEPNAME, args...; kw...)
+
+        function __init__()
+            register(DataDep(
+                DEPNAME,
+                """
+                Dataset: The EMNIST Dataset
+                Authors: Gregory Cohen, Saeed Afshar, Jonathan Tapson, and Andre van Schaik
+                Website: https://www.nist.gov/itl/products-and-services/emnist-dataset
+
+                [Cohen et al., 2017]
+                    Cohen, G., Afshar, S., Tapson, J., & van Schaik, A. (2017).
+                    EMNIST: an extension of MNIST to handwritten letters.
+                    Retrieved from http://arxiv.org/abs/1702.05373
+
+                The EMNIST dataset is a set of handwritten character digits derived from the
+                NIST Special Database 19 (https://www.nist.gov/srd/nist-special-database-19)
+                and converted to a 28x28 pixel image format and dataset structure that directly
+                matches the MNIST dataset (http://yann.lecun.com/exdb/mnist/). Further information
+                on the dataset contents and conversion process can be found in the paper available
+                at https://arxiv.org/abs/1702.05373v1.
+
+                The files are available for download at the official
+                website linked above. Note that using the data
+                responsibly and respecting copyright remains your
+                responsibility. For example the website mentions that
+                the data is for non-commercial use only. Please read
+                the website to make sure you want to download the
+                dataset.
+                """,
+                "http://www.itl.nist.gov/iaui/vip/cs_links/EMNIST/matlab.zip",
+                "e1fa805cdeae699a52da0b77c2db17f6feb77eed125f9b45c022e7990444df95",
+                post_fetch_method = file -> (run(BinDeps.unpack_cmd(file,dirname(file),".zip","")); rm(file))
+            ))
+        end
+
+        module balanced
```

Comment:
To avoid confusion, I think we should stick to the conversion that the module name is uppercased. `EMNIST.Balanced`, `EMNIST.ByMerge`, `EMNIST.ByClass`, `EMNIST.Letters`, `EMNIST.Digits`, and `EMNIST.MNIST`. This looks clearer to me, how do you think?

Reply:
Agreed. The module names have been updated. I have also added some clarification on the front readme page on how the modules are nested for EMNIST since it is sort of 6x as many datasets as the rest. I have also added a table for fast look up of the number of samples (rather than forcing people to go all the way to the EMNIST webpage for that). Does that solve both problems?